Lesson2/Holodenko_Artem/task_2/main.cpp

```diff
@@ -0,0 +1,175 @@
+#include <iostream>
+#include <queue>
+#include <vector>
+#include <math.h>
+
+struct node
+{
+    int n = 0;
+    struct node *left = nullptr;
+    struct node *right = nullptr;
+    struct node *level = nullptr;
+};
+
+
+std::vector<node> get_data()
+{
+    int number_levels, number_nodes=0;
+    std::cout << "Enter number of levels: ";
+    std::cin >> number_levels;
+
+    //кількість вузлів
+    for (int i=0; i<number_levels; i++) number_nodes+=pow(2, i);
+
+    //std::cout << number_nodes << std::endl;
+    std::vector<node> data_vector (number_nodes);
+
+    //заповнення дерева, 0 пустий вузол
+    for (int i=0; i<number_nodes; i++)
+    {
+        std::cout << "Enter node " << i+1 << ": ";
+        std::cin >> data_vector[i].n;
+    }
+
+    //створення зв'язків
+    {
+        int m, n;
+        int level;
+        for (int i=0; i<number_nodes; i++)
+        {
+            level = 0;
+            m=i+1;
+            while (m>pow(2, level))
+            {
+                m-=pow(2, level);
+                level++;
+            }
+            level++;
+
+            //std::cout << "level " << level << std::endl;
+            //std::cout << "m " << m << std::endl;
+
+            n = 0;
+            for (int j=0; j<level; j++) n+=pow(2, j);
```

Comment:
Never write in one line.
Also you should use brackets always, even for one-line statements.